### PR TITLE
Rename GCSReceiver to UAVTalkReceiver

### PR DESCRIPTION
**Pull request by mlyle**
In preparation for pipxtreme injecting it on to-FC telem links.

Also move the initialization into chibi_main/main (with the code
residing in the HAL).

### CODE REVIEW

**Comment by mlyle** [comment]
I need to test to make sure hardware isn't regressed by the reordering-- should be OK, but need to confirm.    Have already confirmed that A) integration test works, and B) GCS receiver control of flightd/sim works.

**Comment by mlyle** [comment]
Tested OK SPRF3E.

**Comment by mlyle** [comment]
Force pushed.  Wasn't based on current next.

**Comment by mlyle** [comment]
Functional on OmnibusF4 when combined with #2162.

File: flight/PiOS/Common/pios_hal.c [hunk 1]
```diff
@@ -1424,3 +1428,21 @@ int PIOS_HAL_ConfigureDAC(dac_dev_t dac)
 	return 0;
 }
 #endif
+
+void PIOS_HAL_InitUAVTalkReceiver()
+{
+#if defined(PIOS_INCLUDE_UAVTALKRCVR)
+	UAVTalkReceiverInitialize();
+	uintptr_t pios_uavtalk_id;
+	PIOS_UAVTALKRCVR_Init(&pios_uavtalk_id);
```
Comment (tracernz) [inline flight/PiOS/Common/pios_hal.c]: Check for failure.

File: flight/PiOS/Common/pios_uavtalkrcvr.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+#include "pios_uavtalkrcvr_priv.h"
+
+#ifndef PIOS_UAVTALKRCVR_TIMEOUT_MS
+#define PIOS_UAVTALKRCVR_TIMEOUT_MS			350
+#endif
+
+static UAVTalkReceiverData uavreceiverdata;
+
+/* Provide a RCVR driver */
+static int32_t PIOS_UAVTALKRCVR_Get(uintptr_t rcvr_id, uint8_t channel);
```
Comment (tracernz) [inline flight/PiOS/Common/pios_uavtalkrcvr.c:46]: Why the mix of caps and lower all through?

Comment (mlyle) [reply flight/PiOS/Common/pios_uavtalkrcvr.c:46]: Existing code.

File: flight/PiOS/Common/pios_hal.c (hunk 1 above)
Comment (mlyle) [reply flight/PiOS/Common/pios_hal.c]: Existing code from all targets :P

Comment (mlyle) [reply flight/PiOS/Common/pios_hal.c]: Fixed though :P

File: flight/PiOS/Common/pios_uavtalkrcvr.c (hunk 2 above)
Comment (tracernz) [reply flight/PiOS/Common/pios_uavtalkrcvr.c:46]: PIOS_UAVTALKRCVR vs PIOS_uavtalkrcvr? :confused:

Comment (mlyle) [reply flight/PiOS/Common/pios_uavtalkrcvr.c:46]: Existing capitalization.  It was just a series of searches and replaces :/  (With a little bit of manual handling of a few bad things already, like some of the RCVR_RCVR's :P)

File: flight/PiOS/Common/pios_uavtalkrcvr.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+/**
+ * Get the value of an input channel
+ * \param[in] channel Number of the channel desired (zero based)
+ * \output PIOS_RCVR_INVALID channel not available
+ * \output PIOS_RCVR_TIMEOUT failsafe condition or missing receiver
+ * \output >=0 channel value
+ */
+static int32_t PIOS_UAVTALKRCVR_Get(uintptr_t rcvr_id, uint8_t channel)
+{
```
Comment (tracernz) [inline flight/PiOS/Common/pios_uavtalkrcvr.c:133]: (existing code) This should really validate the magic.. it's the one "user-space" call that happens regularly after init. But... the channels are stored outside the dev anyway :rofl:

File: flight/PiOS/Common/pios_uavtalkrcvr.c [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+#if defined(PIOS_INCLUDE_UAVTALKRCVR)
+
+#include "pios_uavtalkrcvr_priv.h"
+
+#ifndef PIOS_UAVTALKRCVR_TIMEOUT_MS
+#define PIOS_UAVTALKRCVR_TIMEOUT_MS			350
+#endif
+
+static UAVTalkReceiverData uavreceiverdata;
```
Comment (tracernz) [inline flight/PiOS/Common/pios_uavtalkrcvr.c:43]: (existing code) This belongs inside the device struct, otherwise that struct is basically worthless..?

**Review by tracernz — APPROVED** [review]
Good cleanup. :+1: 

The existing code could do with some help.

File: flight/PiOS/Common/pios_uavtalkrcvr.c (hunk 4 above)
Comment (mlyle) [reply flight/PiOS/Common/pios_uavtalkrcvr.c:43]: Yah, probably.

**Pull request merged**